### PR TITLE
CFY-6090. Write log data to postgresql

### PR DESCRIPTION
**Pull request by jcollado**
In this PR, the logstash configuration is updated to write logs/events to PostgreSQL.

Some comments:
- For now only the message text is stored in PostgreSQL.
  - Logs are written to the `logs` table and events are written to the `events` table.
  - A more complete schema with indexing that implements full text search will be submitted in a separate PR.
- The code assumes that the following resource files are available:
  - Logstash 3rd party output plugin: [`logstash-output-jdbc-0.2.10.gem`](https://rubygems.org/downloads/logstash-output-jdbc-0.2.10.gem)
  - PostgresSQL JDBC driver: [`postgresql-9.4.1212.jar`](https://jdbc.postgresql.org/download/postgresql-9.4.1212.jar)

  My understanding is that these files need to be uploaded to an s3 bucket, but I'm not familiar with the procedure. Any advice on this would be welcome.
- The sanity check has been updated to verity some log/event has been written to PostgreSQL. However the database name has been hardcoded because adding a property to the node didn't work, that is, the property is not available to the script at runtime. Could you let me know what I'm missing in terms of how property values are passed to scripts?

### CODE REVIEW

File: components/logstash/scripts/configure.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+CONFIG_PATH = 'components/logstash/config'
+LOGSTASH_SERVICE_NAME = 'logstash'
+
+
+def configure_logstash():
+
+    logstash_conf_path = '/etc/logstash/conf.d'
+
+    rabbitmq_username = ctx.instance.runtime_properties['rabbitmq_username']
```
Comment (idanmo) [inline components/logstash/scripts/configure.py]: for safety, perhaps use **ctx.instance.runtime_properties.get('rabbitmq_username')**

File: components/logstash/scripts/create.py [hunk 2]
```diff
@@ -71,35 +75,6 @@ def install_logstash():
...
-        '/etc/sysconfig/cloudify-logstash',
-        LOGSTASH_SERVICE_NAME)
-
-    utils.logrotate(LOGSTASH_SERVICE_NAME)
-    utils.sudo(['/sbin/chkconfig', 'logstash', 'on'])
-    utils.clean_var_log_dir(LOGSTASH_SERVICE_NAME)
-
 
-install_logstash()
+if __name__ == '__main__':
```
Comment (idanmo) [inline components/logstash/scripts/create.py]: Why is this needed?
I think that by default these scripts are evaluated by script plugin..

File: components/logstash/scripts/preconfigure_elasticsearch_properties.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    """Get elasticsearch configuration needed to generate logstash.conf."""
+    target = ctx.target
+    source = ctx.source.instance.runtime_properties
+    source['es_endpoint_ip'] = (
+        target.instance.runtime_properties['es_endpoint_ip']
+    )
+    source['es_endpoint_port'] = target.node.properties['es_endpoint_port']
+
+
+if __name__ == '__main__':
```
Comment (idanmo) [inline components/logstash/scripts/preconfigure_elasticsearch_properties.py]: Why is this needed?
I think that by default these scripts are evaluated by script plugin..

File: components/logstash/scripts/preconfigure_postgresql_properties.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        'postgresql_host',
+        'postgresql_db_name',
+        'postgresql_username',
+        'postgresql_password',
+    ]
+    for prop in props:
+        source[prop] = target[prop]
+
+
+if __name__ == '__main__':
```
Comment (idanmo) [inline components/logstash/scripts/preconfigure_postgresql_properties.py]: Why is this needed?
I think that by default these scripts are evaluated by script plugin..

File: components/logstash/scripts/preconfigure_rabbitmq_properties.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    props = [
+        'rabbitmq_endpoint_ip',
+        'rabbitmq_username',
+        'rabbitmq_password',
+    ]
+    for prop in props:
+        source[prop] = target[prop]
+
+
+if __name__ == '__main__':
```
Comment (idanmo) [inline components/logstash/scripts/preconfigure_rabbitmq_properties.py]: Why is this needed?
I think that by default these scripts are evaluated by script plugin..

File: components/manager/scripts/configure_manager.py [hunk 6]
```diff
@@ -92,4 +92,5 @@ def main():
         _disable_requiretty()
     _configure_security_properties()
 
+
```
Comment (idanmo) [inline components/manager/scripts/configure_manager.py]: ?

File: components/manager/scripts/sanity/get_rest_protocol.py [hunk 7]
```diff
@@ -10,4 +10,5 @@ def get_rest_config():
     ctx.source.instance.runtime_properties['rest_protocol'] = rest_protocol
     ctx.source.instance.runtime_properties['rest_port'] = rest_port
 
+
```
Comment (idanmo) [inline components/manager/scripts/sanity/get_rest_protocol.py]: ?

File: components/manager/scripts/sanity/sanity.py [hunk 8]
```diff
@@ -266,6 +278,7 @@ def perform_sanity():
                     'cleaning up sanity resources.')
     _cleanup_sanity()
 
+
```
Comment (idanmo) [inline components/manager/scripts/sanity/sanity.py]: ?

File: components/elasticsearch/scripts/start.py [hunk 9]
```diff
@@ -70,4 +70,5 @@ def main():
         # created in the 'create.py' script.
         es_upgrade_utils.restore_upgrade_data(es_endpoint_ip, ES_ENDPOINT_PORT)
 
+
```
Comment (jcollado) [inline components/elasticsearch/scripts/start.py]: Not sure what version of flake8 is running in the travis CI environment, but as you can see above, the static analysis failed for 4c529cc.

File: components/keepalived/scripts/create.py [hunk 10]
```diff
@@ -19,4 +19,5 @@ def install_keepalived():
     utils.yum_install(keepalived_rpm_url,
                       service_name=KEEPALIVED_SERVICE_NAME)
 
+
```
Comment (jcollado) [inline components/keepalived/scripts/create.py]: Same answer as above, I was just trying to make flake8 in travis CI happy.

File: components/logstash/scripts/create.py (hunk 2 above)
Comment (jcollado) [inline components/logstash/scripts/create.py]: In my opinion is both a matter of style and also keep the ability to test in the future. If the `if __name__ == '__main__'` pattern is not used, then it's not possible to import the script without executing the code.

File: components/logstash/scripts/preconfigure_elasticsearch_properties.py (hunk 3 above)
Comment (jcollado) [inline components/logstash/scripts/preconfigure_elasticsearch_properties.py]: As explained above, this approach is functionally equivalent and allows to test the code in the script in the future if needed.

File: components/logstash/scripts/preconfigure_postgresql_properties.py (hunk 4 above)
Comment (jcollado) [inline components/logstash/scripts/preconfigure_postgresql_properties.py]: Same as above.

File: components/logstash/scripts/preconfigure_rabbitmq_properties.py (hunk 5 above)
Comment (jcollado) [inline components/logstash/scripts/preconfigure_rabbitmq_properties.py]: Same as above.

File: components/manager/scripts/sanity/get_rest_protocol.py (hunk 7 above)
Comment (jcollado) [inline components/manager/scripts/sanity/get_rest_protocol.py]: Also flake8 in travis CI

File: components/logstash/scripts/configure.py (hunk 1 above)
Comment (jcollado) [inline components/logstash/scripts/configure.py]: Good catch. Since there's a conditional below that checks that both `rabbitmq_username` and `rabbitmq_password` are properly set, it makes sense to use `.get` to get a nice error description instead of a key error.

I'll add a new comment once I've verified and pushed the change.

Comment (jcollado) [inline components/logstash/scripts/configure.py]: Fixed in 0d1151d
